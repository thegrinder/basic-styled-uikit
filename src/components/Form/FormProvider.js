import { createProvider } from '../../helpers';

const FormProvider = createProvider('form', {});

export default FormProvider;
