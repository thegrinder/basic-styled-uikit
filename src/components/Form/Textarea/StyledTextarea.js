import styled from 'styled-components';
import { bool } from 'prop-types';
import { commonInputStyles } from '../commonFormStyles';
import { rem } from '../../../theme/typography/typographySelectors';

const propTypes = {
  invalid: bool.isRequired,
};

const StyledTextarea = styled.textarea`
  ${commonInputStyles}
  padding: ${rem(4)} ${rem(10)};
  vertical-align: top;
  overflow: auto;
`;

StyledTextarea.propTypes = propTypes;

export default StyledTextarea;
