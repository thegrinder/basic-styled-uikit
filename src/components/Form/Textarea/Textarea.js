import styled from 'styled-components';
import { bool } from 'prop-types';
import { commonInputStyles } from '../commonFormStyles';
import { rem } from '../../../helpers/utils';

const Textarea = styled.textarea`
  ${commonInputStyles}
  padding: ${rem(4)} ${rem(10)};
  vertical-align: top;
  overflow: auto;
`;

Textarea.propTypes = {
  invalid: bool,
};

Textarea.defaultProps = {
  invalid: false,
};

export default Textarea;
