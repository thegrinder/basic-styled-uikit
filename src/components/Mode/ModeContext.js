import { createContext } from 'react';

const ModeContext = createContext(undefined);

export default ModeContext;
