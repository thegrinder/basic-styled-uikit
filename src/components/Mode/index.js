export { default as ModeProvider } from './ModeProvider';
export { default as useMode } from './useMode';
