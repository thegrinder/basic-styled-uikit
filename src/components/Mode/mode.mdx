---
name: Mode
menu: Components
route: /components/mode
---

## **ModeProvider**



---



## **useMode**
