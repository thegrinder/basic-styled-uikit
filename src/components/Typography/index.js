export { default as TypographyProvider } from './TypographyProvider';
export { default as Text } from './Text/Text';
export * from './typographySelectors';
