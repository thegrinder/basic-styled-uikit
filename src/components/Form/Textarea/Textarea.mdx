---
name: Textarea
menu: Components
---

import { Playground, Props } from 'docz';
import Textarea from './Textarea';


# Textarea

<Props of={Textarea} />


### Textarea

<Playground>
  <Textarea />
</Playground>


### Invalid Textarea

<Playground>
  <Textarea invalid={true} />
</Playground>


### Disabled Textarea

<Playground>
  <Textarea disabled={true} />
</Playground>
