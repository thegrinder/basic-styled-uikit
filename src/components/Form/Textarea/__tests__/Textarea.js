import React from 'react';
import { render } from '@testing-library/react';
import { ModeProvider } from '../../../Mode';
import { TypographyProvider } from '../../../Typography';
import FormProvider from '../../FormProvider';
import Textarea from '../Textarea';

const testId = 'textarea';

const renderComponent = props =>
  render(
    <ModeProvider>
      <FormProvider>
        <TypographyProvider>
          <Textarea data-testid={testId} {...props} />
        </TypographyProvider>
      </FormProvider>
    </ModeProvider>
  );

describe('<Textarea />', () => {
  it('should render correctly with default props', () => {
    const { queryByTestId } = renderComponent();
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('should render correctly with custom props', () => {
    const { queryByTestId } = renderComponent({
      invalid: true,
    });
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('should render <textarea> tag', () => {
    const { queryByTestId } = renderComponent();
    expect(queryByTestId(testId).tagName).toEqual('TEXTAREA');
  });
});
