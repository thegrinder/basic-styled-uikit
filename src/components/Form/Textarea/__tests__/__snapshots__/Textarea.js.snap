// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> should render correctly with custom props 1`] = `
<textarea
  class="StyledTextarea-sc-1v2djou-0 bKoJxV"
/>
`;

exports[`<Textarea /> should render correctly with default props 1`] = `
<textarea
  class="StyledTextarea-sc-1v2djou-0 whAUN"
/>
`;
